portfolio/src/main/webapp/script.js

```diff
@@ -15,14 +15,40 @@
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+let greetingContainer = document.getElementById("greetingContainer");
+let headerNav = document.getElementById("headerNav");
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+
+greetingContainer.addEventListener("mouseover", () => {
+					headerNav.style.opacity = 1;
+				} );
+
+greetingContainer.addEventListener("mouseleave", () => {
+					hideElement(headerNav);
+				});
+// Gradually hides an element.
+function fadeAway(element) {
+	let fadeInterval = setInterval( ()  => {
+  		if (element.style.opacity > 0) {
+ 			element.style.opacity -= .1;
+		} else {
+			clearInterval(fadeInterval);
+		}
+	}, 150);
+}
+
+//Element starts to fade away after 5 seconds.
+function hideElement(element) {
+	setTimeout(() => {fadeAway(headerNav); }, 5000);
+}
```

Comment:
You can actually do exactly this with a little bit of CSS.

```css
#header-nav {
  opacity: 0;
  transition: opacity 0.5s ease 0.5s;
}

#greeting-container:hover #header-nav {
  opacity: 1;
  transition: none; /** show instantly, remove to fade in as well */
}
```

Reply:
Oh wow okay thanks!

Reply:
Is it okay if I actually keep it in JS? I feel like I can modify more easily since I'm more familiar with it.